src/dbt_osmosis/core/osmosis.py

```diff
@@ -1029,6 +1030,31 @@ def remove_columns_not_in_database(
             )
         return changes_committed
 
+    @staticmethod
+    def get_prior_knowledge(
+        knowledge: Dict[str, Dict[str, Any]],
+        column: str,
+    ) -> Dict[str, Any]:
+        camel_column = re.sub("_(.)", lambda m: m.group(1).upper(), column)
+        prior_knowledge_candidates = [
+            knowledge.get(column),
+            knowledge.get(column.lower()),
+            knowledge.get(camel_column),
+        ]
+        sorted_prior_knowledge_candidates_sources = sorted(
+            [knowledge for knowledge in prior_knowledge_candidates if knowledge is not None and knowledge["progenitor"].startswith("source")],
+            key=lambda knowledge: knowledge["generation"],
+            reverse=True,
+        )
+        sorted_prior_knowledge_candidates_models = sorted(
+            [knowledge for knowledge in prior_knowledge_candidates if knowledge is not None and knowledge["progenitor"].startswith("model")],
+            key=lambda knowledge: knowledge["generation"],
+            reverse=True,
+        )
+        sorted_prior_knowledge_candidates = sorted_prior_knowledge_candidates_sources + sorted_prior_knowledge_candidates_models
```

Comment:
If the `generation` is equal, the `source` is preferred. This is because `source` is often the primary information as a column description.

Reply:
I really like this section, looks good